lib/pacemaker/pcmk_sched_actions.c

```diff
@@ -562,34 +572,30 @@ update_action(pe_action_t *then, pe_working_set_t *data_set)
             }
         }
 
-        if (first->rsc && then->rsc && (first->rsc != then->rsc)
-            && (is_parent(then->rsc, first->rsc) == FALSE)) {
+        if ((first->rsc != NULL) && (then->rsc != NULL)
+            && (first->rsc != then->rsc) && !is_parent(then->rsc, first->rsc)) {
             first = action_for_ordering(first);
         }
         if (first != other->action) {
             pe_rsc_trace(then->rsc, "Ordering %s after %s instead of %s",
                          then->uuid, first->uuid, other->action->uuid);
         }
 
-        first_flags = action_flags_for_ordering(first, then_node);
-        then_flags = action_flags_for_ordering(then, first_node);
-
         pe_rsc_trace(then->rsc,
-                     "%s then %s: type=0x%.6x filter=0x%.6x "
-                     "(%s %s %s on %s 0x%.6x then 0x%.6x)",
-                     first->uuid, then->uuid, other->type, first_flags,
-                     action_optional_str(first_flags),
-                     action_runnable_str(first_flags),
-                     action_type_str(first_flags), action_node_str(first),
-                     first->flags, then->flags);
+                     "%s (0x%.6x) then %s (0x%.6x): type=0x%.6x node=%s",
```

Comment:
There's a way to get printf to add the "0x" for you - you can do `(%#.6x)` as the format specifier.

Reply:
It looks like there's a couple dozen other places we could do this same thing, so maybe a separate PR to change them all is in order.

Reply:
Do you know if that is universally supported?

Reply:
It is specified in the standard at least as far back as C99 (ISO9899:1999).